lib/events/complete.go

```diff
@@ -161,6 +171,40 @@ func (u *UploadCompleter) Serve(ctx context.Context) error {
 	}
 }
 
+func (u *UploadCompleter) PerformPeriodicCheck(ctx context.Context) {
+	// If configured with a server ID, then acquire a semaphore prior to completing uploads.
+	// This is used for auth's upload completer and ensures that multiple auth servers do not
+	// attempt to complete the same uploads at the same time.
+	if u.cfg.Semaphores != nil {
+		u.log.Infof("%v: acquiring semaphore in order to complete uploads", u.cfg.ServerID)
```

Comment:
nit: prefer using `slog` for new log lines.

Reply:
I'm not going to address this for now because 
- I don't want to leave this component in a state where half of its logs are slog and half are logrus
- This change should be backported to all supported versions and slog is only on v15+